tracing/tracing/src/main/java/io/helidon/tracing/UnsupportedActivationException.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2024 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.tracing;
+
+/**
+ * Indicates a failed attempt to activate a {@link io.helidon.tracing.Span} due to an error
+ * in a {@link io.helidon.tracing.SpanLifeCycleListener} callback.
+ * <p>
+ *     Callers should normally catch this exception and then retrieve and close the {@link io.helidon.tracing.Scope}.
+ *     Helidon activates the scope before invoking span life cycle listeners, so the span and baggage have already
+ *     been added to the current tracing context. In the absence of this exception being thrown, the
+ *     {@link io.helidon.tracing.Span#activate()} method returns the {@code Scope} so the caller can close it. But
+ *     when Helidon throws this exception due to an error in a listener, the caller has no access to the {@code Scope} return value
+ *     return value.
+ */
+public class UnsupportedActivationException extends UnsupportedOperationException {
```

Comment:
Controversial opinion: consider implementing `AutoCloseable` directly.

Reply:
`UnsupportedOperationException` is designed for collection that do not support a subset of operations (such as write operations on a read only collection).
I do not think it is relevant for this use case, as all operations must be (by design of the API) supported. 
I think extending `RuntimeException` is more aligned with what this does